pkgs/development/python-modules/pyradiomics/default.nix

```diff
@@ -0,0 +1,71 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, pytestCheckHook
+, numpy
+, pykwalify
+, pywavelets
+, setuptools
+, simpleitk
+, six
+, versioneer
+}:
+
+buildPythonPackage rec {
+  pname = "pyradiomics";
+  version = "3.1.0";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "AIM-Harvard";
+    repo = "pyradiomics";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-/qFNN63Bbq4DUZDPmwUGj1z5pY3ujsbqFJpVXbO+b8E=";
+    name = "pyradiomics";
+  };
+
+  nativeBuildInputs = [ setuptools versioneer ];
+
+  propagatedBuildInputs = [
+    numpy
+    pykwalify
+    pywavelets
+    simpleitk
+    six
+  ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+  preCheck = ''
+    rm -rf radiomics
+  '';
+  # tries to access network at collection time:
+  disabledTestPaths = [ "tests/test_wavelet.py" ];
+  # various urllib download errors and (probably related) missing feature errors:
+  disabledTests = [
+    "brain1_shape2D-original_shape2D"
+    "brain2_shape2D-original_shape2D"
+    "breast1_shape2D-original_shape2D"
+    "lung1_shape2D-original_shape2D"
+    "lung2_shape2D-original_shape2D"
+  ];
+  # note the above elements of disabledTests are patterns, not exact tests,
+  # so simply setting `disabledTests` does not suffice:
+  pytestFlagsArray = [
+    "-k '${toString (lib.intersperse "and" (lib.forEach disabledTests (t: "not ${t}")))}'"
+  ];
```

Comment:
Hi! I'm authoring a tree-wide change to deprecate `pytestFlagsArray` with `(enabled|disabled)(TestPaths|TestMarks|Tests)` and `pytestFlags`.

What does it mean by saying those elements are "patterns" instead of "exact tests"? The `pytestCheckHook` has been using `-k "not test1 or not test2"` to handle `disabledTests` even before the introduction of `pytestFlags`.

Reply:
Sorry, I can't remember exactly what I was thinking, maybe I tried using disabledTests and misdiagnosed some issue at the time? Doesn't seem worth worrying about now, probably.